logging/src/main/java/org/apache/rocketmq/logging/inner/LoggingBuilder.java

```diff
@@ -708,17 +708,30 @@ public synchronized void setFile(String fileName, boolean append, boolean buffer
                     throw ex;
                 }
             }
-            Writer fw = createWriter(ostream);
-            if (bufferedIO) {
-                fw = new BufferedWriter(fw, bufferSize);
+            Writer fw = null;
+            try {
+                fw = createWriter(ostream);
+                if (bufferedIO) {
+                    fw = new BufferedWriter(fw, bufferSize);
+                }
+                this.setQWForFiles(fw);
+                this.fileName = fileName;
+                this.fileAppend = append;
+                this.bufferedIO = bufferedIO;
+                this.bufferSize = bufferSize;
+                writeHeader();
+                SysLogger.debug("setFile ended");
+            } catch (Exception e) {
+                throw e;
+            } finally {
+                if (ostream != null) {
+                    ostream.close();
+                }
+                if (fw != null) {
```

Comment:
The osstream and fw is referenced by the qwWriter, which will be released via coseFile().

Reply:
the ostream and fw cannot be closed here, otherwise, the logger appender will fail to write the log.

Reply:
it is better to reproduce the issue and add some tests.

Reply:
Thank you very much for your advice

I tried it and did get an error when printing logs asynchronously, so I turned off pr for now